File: server/internal/model/paste.go
package model

import (
	"time"

	"github.com/jackc/pgx/v5/pgtype"
)

type Paste struct {
	Slug      string             `json:"slug"`
	Content   string             `json:"content"`
	ExpiresAt pgtype.Timestamptz `json:"expires_at"`
}

type CreatePasteRequest struct {
	Content    string  `json:"content"`
	Expiration string  `json:"expiration"` // "1h", "1d", etc.
	Password   *string `json:"password"`
}

type CreatePasteResponse struct {
	Slug      string    `json:"slug"`
	ExpiresAt time.Time `json:"expires_at"`
}

type CreateGetResponse struct {
	Content   string    `json:"content"`
	ExpiresAt time.Time `json:"expires_at"`
}

File: client/src/lib/types/paste.ts
export interface CreatePasteRequest {
	content: string;
	expiration: string;
	password?: string;
}

export interface CreatePasteResponse {
	ok: true;
	status: 200;
	slug: string;
}

export interface GetPasteResponse {
	ok: true;
	status: 200;
	content: string;
	expires_at: string;
}

export interface Paste {
	slug: string;
	content: string;
	expires_at: string;
}
